credentials/src/test/kotlin/SSITest.kt

```diff
@@ -124,10 +161,29 @@ class SSITest {
 
     val vcJwt: VcJwt = VerifiableCredential.create(signOptions, null, vc)
 
-    val createVpOptions = CreateVpOptions(getPresentationDefinition(), arrayListOf(vcJwt), did)
+    val btcAddressPd = buildPresentationDefinition(
+      inputDescriptors = listOf(
+        buildInputDescriptor(fields = listOf(buildField(paths = arrayOf("$.credentialSubject.btcAddress"))))
+      )
+    )
+
+    val createVpOptions =
+      CreateVpOptions(btcAddressPd, arrayListOf(vcJwt), did)
     val vpJwt: VpJwt = VerifiablePresentation.create(signOptions, createVpOptions)
 
     assertTrue(VerifiablePresentation.verify(vpJwt, SimpleResolver(didDocument)))
+
+    val parts = vpJwt.split(".")
+    val header = String(Base64.getDecoder().decode(parts[0]))
+    val payload = String(Base64.getDecoder().decode(parts[1]))
+
+    // Header Checks
+    assertTrue { header.contains("\"alg\":\"") }
+    assertTrue { header.contains("\"typ\":\"JWT\"") }
+
+    // Payload Checks
+    assertTrue { payload.contains("\"iss\":\"") }
+    assertTrue { payload.contains("\"sub\":\"") }
```

Comment:
Seems strange to me to not use the nimbus classes. What's the motivation here?

Reply:
Quick and dirty validation, 
switched to nimbus! 👍